File: src/frontend/src/backend/roomService.js
import {config, appFetch} from './appFetch';

export const addRoom = (room, onSuccess, onErrors) => 
    appFetch(`/hotels/${room.hotel.id}/rooms`, config('POST', room), onSuccess, onErrors);

export const findRoom = (hotelid, roomid, onSuccess, onErrors) =>
    appFetch(`/hotels/${hotelid}/rooms/${roomid}`, config('GET'), onSuccess, onErrors);

export const findRooms = ({hotelid, status, type,page}, onSuccess, onErrors) => {
    let path = `/hotels/${hotelid}/rooms/?page=${page}`;

    path+= status ? `&status=${status}` : "";

    path+= type ? `&type=${type}` : "";

    appFetch(path, config('GET'), onSuccess, onErrors);
}

export const findAllRoomTypes = (onSuccess) => 
    appFetch('/hotels/roomtypes', config('GET'), onSuccess);


export const updateRoom = (room, onSuccess, onErrors) => 
    appFetch(`/hotels/${room.hotel.id}/rooms/${room.id}`, config('PUT', room), onSuccess, onErrors);

export const removeRoom = (hotelid, roomid, onSuccess, onErrors) => 
    appFetch(`/hotels/${hotelid}/rooms/${roomid}`, config('DELETE'), onSuccess, onErrors);

File: src/backend/src/main/java/es/udc/hotelapp/backend/model/services/RoomService.java
package es.udc.hotelapp.backend.model.services;

import java.util.List;

import es.udc.hotelapp.backend.model.entities.Room;
import es.udc.hotelapp.backend.model.entities.RoomType;
import es.udc.hotelapp.backend.model.exceptions.InstanceNotFoundException;
import es.udc.hotelapp.backend.model.exceptions.RoomAlreadyExistsException;

public interface RoomService {
	
	public Long addRoom(Room room) throws RoomAlreadyExistsException, InstanceNotFoundException;

	public void updateRoom(Room r1);

	public void removeRoom(Room r1) throws InstanceNotFoundException;

	public Room findRoom(Long id) throws InstanceNotFoundException;

	public List<Room> findRooms(String status, Long hotelid, String type);
	
	public List<RoomType> findAllRoomTypes();
}
